### PR TITLE
improvement/S3C-2749 add tests for url duration

### PR DESCRIPTION
**Pull request by ilkescality**
Adds unit tests for customizing S3 pre-signed URL implemented here: #970

### CODE REVIEW

File: tests/unit/auth/v2/queryAuthCheck.js [hunk 1]
```diff
@@ -26,3 +27,94 @@ describe('v2: queryAuthCheck', () => {
...
 });
+
+describe('v2: queryAuthCheck', () => {
+    let clock;
+
+    beforeEach(() => {
+        clock = sinon.useFakeTimers();
+    });
+    afterEach(() => {
+        process.env.PRE_SIGN_URL_EXPIRY = 604800000;
```
Comment (naren-rajendran) [inline tests/unit/auth/v2/queryAuthCheck.js:38]: I think I missed it in your original PR, could you add the default expiry value to  `/arsenal/lib/constants.js` and use that in `queryAuthCheck.js` and here?  You can include the change in this PR I guess!

File: package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
     "level": "~5.0.1",
     "level-sublevel": "~6.6.5",
     "node-forge": "^0.7.1",
-    "simple-glob": "^0.1",
+    "simple-glob": "^0.2.0",
```
Comment (dora-korpar) [inline package.json]: Are these package updates needed for the tests? Cos if not they should be in a separate PR

File: tests/unit/auth/v2/queryAuthCheck.js (hunk 1 above)
Comment (jonathan-gramain) [reply tests/unit/auth/v2/queryAuthCheck.js:38]: If you are going to be adding this, *optionally* you might consider changing the default unit to seconds and convert it to the required unit, it's a more familiar unit and less 0s to count :wink:

Comment (jonathan-gramain) [reply tests/unit/auth/v2/queryAuthCheck.js:38]: Only do this if it's an easy change I don't want you to break all the stack for this :)

File: tests/unit/auth/v2/queryAuthCheck.js [hunk 3]
```diff
@@ -26,3 +27,94 @@ describe('v2: queryAuthCheck', () => {
         }
     }));
 });
+
+describe('v2: queryAuthCheck', () => {
+    let clock;
+
+    beforeEach(() => {
+        clock = sinon.useFakeTimers();
```
Comment (jonathan-gramain) [inline tests/unit/auth/v2/queryAuthCheck.js:35]: :+1:

File: package.json (hunk 2 above)
Comment (ilkescality) [reply package.json]: They shouldn't be needed (unless sinon has dependency for their specific version which I don't think is the case), I'll try only leaving the new package and revert `simple-glob` and `eslint-plugin-react` to their original versions.

File: tests/unit/auth/v2/queryAuthCheck.js (hunk 1 above)
Comment (nicolas2bert) [inline tests/unit/auth/v2/queryAuthCheck.js:38]: Minor: for better readability, you might want to add a global variable; something like `const SEVEN_HOURS_IN_SEC = 7 * 24 * 60 * 60`;

File: tests/unit/auth/v2/queryAuthCheck.js [hunk 4]
```diff
@@ -26,3 +27,94 @@ describe('v2: queryAuthCheck', () => {
...
+        assert.notStrictEqual(res.err.AccessDenied, true);
+        assert.notStrictEqual(res.err.RequestTimeTooSkewed, true);
+    });
+    it('URL should expire after 7 days with default expiry', () => {
+        clock.tick(604800000);  // take time 604800000ms (7 days) ahead
+        const currentTime = Date.now();
+        const request = { method: 'GET', query: { Expires: currentTime } };
+        const data = { Expires: currentTime };
+        const res = queryAuthCheck(request, log, data);
+        assert.notStrictEqual(res.err, undefined);
```
Comment (jonathan-gramain) [inline tests/unit/auth/v2/queryAuthCheck.js:70]: minor style: might be slightly more readable to do `assert(res.err)` and it does a truish test, as some callbacks return `null` as the error object (not `undefined`), it can be a more generic way to check that an error occurred.

File: tests/unit/auth/v2/queryAuthCheck.js (hunk 1 above)
Comment (ilkescality) [reply tests/unit/auth/v2/queryAuthCheck.js:38]: Sure, will try that and see if it breaks anything, so far it doesn't seem like it would, but I'll double-check. :)

**Review by jonathan-gramain — APPROVED** [review]

**Review by nicolas2bert — APPROVED** [review]

File: tests/unit/auth/v2/queryAuthCheck.js (hunk 1 above)
Comment (naren-rajendran) [reply tests/unit/auth/v2/queryAuthCheck.js:38]: IMHO, you should, makes it easier to set!

Comment (ilkescality) [reply tests/unit/auth/v2/queryAuthCheck.js:38]: Good idea

**Comment by naren-rajendran** [comment]
@ilkescality I think you should squash your commits.

File: lib/constants.js [hunk 5]
```diff
@@ -72,4 +72,7 @@ module.exports = {
     permittedCapitalizedBuckets: {
         METADATA: true,
     },
+    // Default expiration value of the S3 pre-signed URL duration
+    // 604800 seconds (seven days).
+    defaultExpiry: 7 * 24 * 60 * 60,
```
Comment (naren-rajendran) [inline lib/constants.js]: May be a more descriptive name? `defaultPreSignedURLExpiry` instead of `defaultExpiry`!

File: tests/unit/auth/v2/queryAuthCheck.js (hunk 4 above)
Comment (ilkescality) [reply tests/unit/auth/v2/queryAuthCheck.js:70]: TBH, it bothered me as well :) I thought it could be good to have a helper util such as `nullOrUndefined` or something like that - maybe another to check for even more falsey values depending on the use case... For now, I added additional check for the `null` case to cover more potential scenarios.

Comment (naren-rajendran) [reply tests/unit/auth/v2/queryAuthCheck.js:70]: May be I am wrong, I don't see `queryAuthCheck` returning `err: undefined`!

**Comment by ilkescality** [comment]
@bert-e force_reset

**Review by rachedbenmustapha — APPROVED** [review]
